packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -146,6 +146,57 @@ class HotRunner extends ResidentRunner {
     throw 'Failed to compile $expression';
   }
 
+  @override
```

Comment:
It would be good to have some comments here or in the base class explaining that this is the implementation of a service protocol extension. The comment should also discuss pre- and post-conditions for calling the extension. What should be true about the library and the class whose IDs get passed here? What should happen after this call to make the whole thing do something?

Reply:
Done